pasha/context.py

```diff
@@ -39,56 +39,154 @@ def __init__(self, num_workers):
 
         self.num_workers = num_workers
 
-    def array(self, shape, dtype=np.float64):
-        """Allocate an array shared with all workers.
+    @staticmethod
+    def _resolve_alloc(shape, dtype, order, like):
+        """Resolve allocation arguments.
 
-        The implementation may decide how to back this memory, but it
-        is required that all workers of this context may read AND write
-        to this memory. The base implementation allocates directly
-        on the heap.
+        This method resolves the arguments passed to
+        :method:`MapContext.alloc` and fills any omitted values.
+
+        Returns:
+            (tuple, DtypeLike, str): Resolved shape, data-type and
+                memory order.
+        """
+        if like is not None:
+            shape = shape or like.shape
```

Comment:
Bit surprising if you pass `shape=0` :wink:

Reply:
Hu, I knew I should've sticked with the recommended `x is None` checks rather than cutting it short 🙄

Reply:
I do like how `arg = arg or default` reads, and there are places where you can get away with it. But yeah, there's a reason the `is None` check is recommended. :slightly_smiling_face: